packages/webhooks/lib/utils.ts

```diff
@@ -27,12 +29,39 @@ export const NON_FORWARDABLE_HEADERS = [
 
 export const retry = async (logCtx?: LogContext | null, error?: AxiosError, attemptNumber?: number): Promise<boolean> => {
     if (error?.response && (error?.response?.status < 200 || error?.response?.status >= 300)) {
-        const content = `Webhook response received an ${
+        const content = `Webhook response received a ${
             error?.response?.status || error?.code
         } error, retrying with exponential backoffs for ${attemptNumber} out of ${RETRY_ATTEMPTS} times`;
 
-        await logCtx?.error(content);
+        const meta: Record<string, unknown> = {};
 
+        if (error.code) {
+            meta['code'] = error.code;
+        }
+
+        if (error.response) {
+            const metaResponse: Record<string, unknown> = {};
+            metaResponse['status'] = error.response.status;
+            if (error.response.headers) {
+                metaResponse['headers'] = redactHeaders({ headers: error.response.headers });
+            }
+            if (error.response.data) {
+                metaResponse['data'] = error.response.data;
+            }
+            meta['response'] = metaResponse;
+
+            if (error.request) {
+                const request = error.request as ClientRequest;
+
+                const metaRequest: Record<string, unknown> = {};
+                metaRequest['url'] = `${request.protocol}//${request.host}${request.path}`;
+                metaRequest['method'] = request.method;
+                metaRequest['headers'] = redactHeaders({ headers: request.getHeaders() });
+                meta['request'] = metaRequest;
+            }
+        }
+
+        await logCtx?.error(content, truncateJson(meta));
```

Comment:
there is a convenience method `logCtx.http()` to log http request.

Reply:
@TBonnin Actually I'm not in love with that method because it doesn't write the response body, which could be helpful in these cases. Would we want to add that?

Reply:
Good point, I think we can definitely add a new field to the `type: http`, alternatively you can use `http()` and log the body in the meta field freely. One of the goal with this method is to have a std way to log HTTP across the codebase, let's try to make that fit 💪🏻

Reply:
Added body. I'm not going to mess with it here, but I really wish the `code` field on those http response logs was `status` or `status_code` instead.